src/Aquifer.AI/OpenAiTranslationService.cs

```diff
@@ -154,6 +150,42 @@ public async Task<string> TranslateHtmlAsync(
         return translatedHtml.ToString();
     }
 
+    private static (string Text, bool IsFullReplace) ReplaceTranslationPairs(string text, IDictionary<string, string> translationPairs)
+    {
+        foreach (var pair in translationPairs.OrderByDescending(x => x.Key.Length))
+        {
+            if (string.Equals(pair.Key, text, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return (pair.Value, true);
+            }
+
+            text = Regex.Replace(text, $"""\b(?:{pair.Key})\b""", pair.Value, RegexOptions.IgnoreCase);
+        }
+
+        return (text, false);
+    }
+
+    private async Task<string> TranslateTextAsync(string prompt, string text, CancellationToken cancellationToken)
```

Comment:
This could probably be merged into the method below but I'm not going to worry about that right now.